snuba_sdk/query_visitors.py

```diff
@@ -204,6 +211,8 @@ def _combine(self, query: main.Query, returns: Mapping[str, str]) -> str:
             "dry_run",
             "legacy",
             "parent_api",
+            "team",
```

Comment:
This check is a legit gotcha if you aren't aware of it. I think the way the Query stores flags needs to be refactored to make the logic easier to understand.